pkg/diff/sql_generator.go

```diff
@@ -402,18 +447,27 @@ func (schemaSQLGenerator) Alter(diff schemaDiff) ([]Statement, error) {
 		return nil, fmt.Errorf("resolving renaming conflicting indexes: %w", err)
 	}
 
-	indexSQLVertexGenerator := indexSQLVertexGenerator{
+	indexGraphs, err := diff.indexDiffs.resolveToSQLGraph(&indexSQLVertexGenerator{
 		deletedTablesByName:      deletedTablesByName,
-		addedTablesByName:        buildSchemaObjByNameMap(diff.tableDiffs.adds),
+		addedTablesByName:        addedTablesByName,
 		tablesInNewSchemaByName:  tablesInNewSchemaByName,
 		indexesInNewSchemaByName: buildSchemaObjByNameMap(diff.new.Indexes),
 		indexRenamesByOldName:    renameConflictingIndexSQLVertexGenerator.getRenames(),
-	}
-	indexGraphs, err := diff.indexDiffs.resolveToSQLGraph(&indexSQLVertexGenerator)
+	})
 	if err != nil {
 		return nil, fmt.Errorf("resolving index sql graphs: %w", err)
 	}
 
+	fkConsGraphs, err := diff.foreignKeyConstraintDiffs.resolveToSQLGraph(&foreignKeyConstraintSQLVertexGenerator{
+		deletedTablesByName:           deletedTablesByName,
+		addedTablesByName:             addedTablesByName,
```

Comment:
double checking: Will this include recreated tables?

Reply:
Yes! We validate this works in the test cases that change a table from partitioned to unpartitioned and vice-versa.